tests/pytests/functional/states/test_npm.py

```diff
@@ -0,0 +1,39 @@
+import pytest
+from salt.exceptions import CommandExecutionError
+
+
+@pytest.fixture(scope="module", autouse=True)
+def install_npm(sminion):
+    try:
+        sminion.functions.pkg.install("npm")
+        # Just name the thing we're looking for
+        sminion.functions.npm  # pylint: disable=pointless-statement
+    except (CommandExecutionError, AttributeError):
+        pytest.skip("Unable to install npm")
+
+
+@pytest.mark.slow_test
+@pytest.mark.destructive_test
+@pytest.mark.requires_network
+def test_removed_installed_cycle(sminion):
+    project_version = "pm2@5.1.0"
+    success = sminion.functions.npm.uninstall("pm2")
+    assert success, "Unable to uninstall pm2 in prep for tests"
+
+    ret = next(
+        iter(
+            sminion.functions.state.single(
+                "npm.installed", name=project_version
+            ).values()
+        )
+    )
+    success = ret["result"]
+    assert success, "Failed to states.npm.installed " + project_version + ret["comment"]
+
+    ret = next(
+        iter(
+            sminion.functions.state.single("npm.removed", name=project_version).values()
+        )
+    )
+    success = ret["result"]
+    assert success, "Failed to states.npm.removed " + project_version
```

Comment:
```suggestion
def test_removed_installed_cycle(npm, npm_mod):
    project_version = "pm2@5.1.0"
    success = npm_mod.uninstall("pm2")
    assert success, "Unable to uninstall pm2 in prep for tests"

    ret = npm.installed(name=project_version)
    assert ret.result is True

    ret = npm.removed(name=project_version)
    assert ret.result is True
```
`sminion` should only be used for prep tasks which would otherwise require a minion and where salt can help.
For tests, use the `modules` fixture to access salt modules and the `states` fixture to access state modules.

Reply:
Why do we need to define npm/npm_mod fixtures? autouse on the install fixture handles ensuring that it's installed, and I'm assuming that states & modules fixtures are already correctly scoped?

Reply:
`npm`, and `npm_mod` are shortcuts so that you don't have to use `states.npm` and `modules.npm` which, yes, they are correctly scoped

Reply:
I'm fine using states.npm and modules.npm -- if I were *only* using the module/state and not any other bits that could be convenient, but states/modules is short enough that it's not that much of an inconvenience... for me :joy:

Reply:
(also I prefer the explicit use of states/modules... which *could* be 'npm_state' and 'npm_mod' but... then I'm basically right where I am now :trollface: )

Reply:
It's just a shortcut, not a required use :)
So, whichever you prefer applies perfectly.